### ISSUE TITLE
[Feature Request] Set Remote WSL as default run option without depending on the workspace

### ISSUE THREAD

**Issue by gusbemacbe** [microsoft/vscode-remote-release#2365]
Hello!

- I know that the workspace can do it, but even if I set `"window.restoreWindows": "one"` and `"window.restoreWindows": "all"`, it opens automatically the Remote WSL, but if you open any file or any folder, it will open the Remote WSL, but not with any clicked file or folder.
- I know that it is possible to open VSCode with Remote WSL automatically from the WSL terminal, but I want to open VSCode with Remote WSL automatically and normally **_without from_** terminal, with any file or any folder. 

I would like to suggest that make it possible to set Remote WSL as default run option when I open VSCode. Referring to the subreddit topic [Can I make Remote WSL as default run option?](https://www.reddit.com/r/vscode/comments/eak5t3/can_i_make_remote_wsl_as_default_run_option/).

### References

- https://github.com/microsoft/WSL/issues/4127
- https://github.com/microsoft/vscode/issues/3884

**Comment by DachuanZhao** [comment]
This is a very useful feature because I only install python in wsl2 . I have to change to windows terminal , enter the directory and ```code xxx.py``` when I want to open and run a python file in windows .

**Comment by sidharthramesh** [comment]
Yes. This will be a very useful feature to have. Just having the file and folder context option "Open with Code" recognize the user is opening at a path like `\\wsl$\Ubuntu\home\` and opening VSCode with Remote WSL would be an awesome feature to have.

**Comment by sidharthramesh** [comment]
So, I've made a go script that will automatically detect the path and launch VSCode smartly using "Open with Code". 

```golang
package main

import (
	"fmt"
	"os"
	"os/exec"
	"strings"
)

func convertPath(path string) string {
	formatted := strings.Trim(path, "\\\\wsl$\\Ubuntu")
	formatted = strings.ReplaceAll(formatted, "\\", "/")
	return formatted
}

func main() {
	path := os.Args[1]
	localPath := os.Getenv("LOCALAPPDATA")
	var finalArguments []string
	if strings.HasPrefix(path, "\\\\wsl$") {
		finalArguments = append(finalArguments, "--remote", "wsl+ubuntu", convertPath(path))
	} else {
		finalArguments = append(finalArguments, path)
	}
	cmd := exec.Command(fmt.Sprintf("%v\\Programs\\Microsoft VS Code\\code.exe", localPath), finalArguments...)
	cmd.Run()
}
```
I built it using `go build -ldflags -H=windowsgui` to hide the terminal window. You can get the exe file [here](https://github.com/sidharthramesh/openwithcode/blob/master/opencode.exe).
Just edit the following registries
- `Computer\HKEY_CLASSES_ROOT\Directory\Background\shell\VSCode\command` -> `"C:\users\path\to\opencode.exe" "%V"`
- `Computer\HKEY_CLASSES_ROOT\Directory\shell\VSCode\command`  -> `"C:\users\path\to\opencode.exe" "%V"`

**Comment by gusbemacbe** [comment]
@sbatten, please, can you re-open it?

**Comment by aeschli** [comment]
We currently have no plans to work on the suggested feature. Please have a look at the solution proposed in https://github.com/microsoft/vscode-remote-release/issues/2365#issuecomment-695937925

**Comment by mrlami** [comment]
now that it has more than 11 upvotes, is this something that can make it to your backlog?
